consul/resource_consul_config_entry_v2_exported_services.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package consul
+
+import (
+	"encoding/json"
+	"fmt"
+
+	pbmulticluster "github.com/hashicorp/consul/proto-public/pbmulticluster/v2"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+func resourceConsulV2ExportedServices() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceConsulV2ExportedServicesCreate,
+		Update: resourceConsulV2ExportedServicesUpdate,
+		Read:   resourceConsulV2ExportedServicesRead,
+		Delete: resourceConsulV2ExportedServicesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the config entry to read.",
+			},
+
+			"kind": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The kind of exported services config (ExportedServices, NamespaceExportedServices, PartitionExportedServices).",
+			},
+
+			"partition": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The partition the config entry is associated with.",
+				ForceNew:    true,
+			},
+
+			"namespace": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The namespace the config entry is associated with.",
+				ForceNew:    true,
+			},
+
+			"services": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported services.",
+			},
+
+			"partition_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service partition consumers.",
+			},
+			"peer_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service peer consumers.",
+			},
+			"sameness_group_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service sameness group consumers.",
+			},
+		},
+	}
+}
+
+func resourceConsulV2ExportedServicesCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceConsulV2ExportedServicesUpdate(d, meta)
+}
+
+func resourceConsulV2ExportedServicesUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, _, wOpts := getClient(d, meta)
+	name := d.Get("name").(string)
+	kind := d.Get("kind").(string)
+	gvk := &GVK{
+		Group:   "multicluster",
+		Version: "v2",
+		Kind:    kind,
+	}
+	var consumers []map[string]any
+	peerConsumers := d.Get("peer_consumers").([]interface{})
```

Comment:
All the Get calls with `d.Get` will fail of user has not given the key in schema. From the documentation 
```
// Get returns the data for the given key, or nil if the key doesn't exist
// in the schema.
```
Since these are optional keys we need to add nil checks everywhere.

Reply:
refactored to in-line the loop - this pattern is used with optional keys everywhere